### PR TITLE
Enable CLI plugins in oc

### PR DESCRIPTION
**Pull request by fabianofranz**
Trello story: https://trello.com/c/o6UiTkxv

Adds support to CLI plugins under the `oc plugin` command. Basically the same functionality and same subcommand already present upstream. 

This is required for enabling the service catalog commands under `oc`, which will be distributed as plugins.

@liggitt one upstream commit for a feature that didn't make the 1.7 cut upstream
@openshift/cli-review @smarterclayton prease review
@sspeiche @pmorie FYI

### CODE REVIEW

**Comment by juanvallejo** [comment]
flaked on https://github.com/openshift/origin/issues/8571
/retest

**Comment by danwinship** [comment]
> sorry, I don't know why this was assigned to me or how I can help with it

yeah, likewise. (well, I know it was assigned to me because this includes tests and i'm listed as an approver in some subdirs of test/, but that seems like a bug in the assigning mechanism)

/unassign

File: vendor/k8s.io/kubernetes/pkg/kubectl/plugins/plugins.go [hunk 1]
```diff
@@ -71,3 +81,33 @@ func (p Plugin) IsValid() bool {
 
 // Plugins is a list of plugins.
 type Plugins []*Plugin
+
+// Flag describes a single flag supported by a given plugin.
+type Flag struct {
```
Comment (juanvallejo) [inline vendor/k8s.io/kubernetes/pkg/kubectl/plugins/plugins.go]: @fabianofranz would it be worth having a struct for this?

**Comment by juanvallejo** [comment]
One minor comment, but otherwise this lgtm from a cli perspective

File: vendor/k8s.io/kubernetes/pkg/kubectl/plugins/plugins.go (hunk 1 above)
Comment (fabianofranz) [inline vendor/k8s.io/kubernetes/pkg/kubectl/plugins/plugins.go]: I couldn't think of any use case where there would be different implementations of these guys, so I don't think it's worth at the moment.  ;)

Comment (juanvallejo) [inline vendor/k8s.io/kubernetes/pkg/kubectl/plugins/plugins.go]: Sounds good to me

**Comment by fabianofranz** [comment]
@smarterclayton @liggitt UP

**Pull request merged**